solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -151,8 +173,15 @@ contract WalletRegistry is Ownable {
 
         dkg.lockState();
         dkg.start(
-            uint256(keccak256(abi.encodePacked(relayEntry, block.number)))
+            uint256(keccak256(abi.encodePacked(randomRelayEntry, block.number)))
```

Comment:
It may be possible that subsequent DKG executions are started with the same `randomRelayEntry`, (e.g. when the beacon doesn't produce a new relay entry). We should start DKG with a unique seed anyways, so we're adding the `block.number`.

Reply:
Ok, this comment also depends on [the question](https://github.com/keep-network/keep-core/pull/2801#discussion_r795657136) of whether we can create a new wallet when the random beacon fails to provide a new entry. Unless you're sure that we can do that?

Reply:
We can't use the previous entry/block number to select operators to a new wallet, it increases the attack surface. Relay entry must not be known at the time a new wallet is requested.